internal/provider/burn_alert_resource.go

```diff
@@ -97,10 +98,25 @@ func (*burnAlertResource) Schema(_ context.Context, _ resource.SchemaRequest, re
 				},
 			},
 			"dataset": schema.StringAttribute{
-				Required:    true,
+				Optional:    true,
+				Computed:    true,
 				Description: "The dataset this Burn Alert is associated with.",
 				PlanModifiers: []planmodifier.String{
-					stringplanmodifier.RequiresReplace(),
+					modifiers.DatasetDeprecation(),
```

Comment:
final thing: can we get some test coverage for the cases where there planmodifiers run?